packages/core-sdk/src/resources/dispute.ts

```diff
@@ -19,102 +22,240 @@ export class DisputeClient {
     }
   }
-  /*
-  public async raiseDispute(request: RaiseDisputeRequest): Promise<RaiseDisputeResponse> {
+
+  /**
+   * Resolves an existing dispute after a judgement has been set.
+   * @param request The request object containing details to resolve the dispute.
+   *   @param request.disputeId The ID of the dispute to be resolved.
+   * @returns A Promise that resolves to a ResolveDisputeResponse.
+   * @throws NotAbleToResolve, if currentTag is still in dispute (i.e still needs a judgement to be set)
+   * @throws NotDisputeInitiator, if the transaction executor is not the one that initiated the dispute
+   * @emits DisputeResolved (_disputeId)
+   */
+  public async resolveDispute(request: ResolveDisputeRequest): Promise<ResolveDisputeResponse> {
     try {
       const { request: call } = await this.rpcClient.simulateContract({
         ...DisputeModuleConfig,
-        functionName: "raiseDispute",
-        args: [
-          request.targetIpId,
-          request.arbitrationPolicy,
-          request.linkToDisputeEvidence,
-          stringToHex(request.targetTag, { size: 32 }),
-          request.calldata ? request.calldata : "0x",
-        ],
+        functionName: "resolveDispute",
+        args: [BigInt(request.disputeId)],
       });
 
       const txHash = await this.wallet.writeContract(call);
 
+      if (request.txOptions?.waitForTransaction) {
+        await waitTxAndFilterLog(this.rpcClient, txHash, {
+          ...DisputeModuleConfig,
+          eventName: "DisputeResolved",
+        });
+      }
+
       return { txHash: txHash };
     } catch (error) {
-      handleError(error, "Failed to raise dispute");
+      handleError(error, "Failed to cancel dispute");
     }
   }
-*/
-  public async setDisputeJudgement(
-    request: SetDisputeJudgementRequest,
-  ): Promise<SetDisputeJudgementResponse> {
-    try {
-      const { request: call } = await this.rpcClient.simulateContract({
-        ...DisputeModuleConfig,
-        functionName: "setDisputeJudgement",
-        args: [
-          BigInt(request.disputeId),
-          request.decision,
-          request.calldata ? request.calldata : "0x",
-        ],
-      });
 
-      const txHash = await this.wallet.writeContract(call);
+  /* Read functions
+			- These functions are used to read storage variables from the contract
+			- If you want to read events data emitted, check out our indexer API documentation
+	*/
 
-      return { txHash: txHash };
-    } catch (error) {
-      handleError(error, "Failed to raise dispute");
-    }
+  /**
+   * Reads from the contract's disputes mapping storage variable.
+   * @param disputeId string | number | bigint, The ID of the dispute to be read.
+   * @returns The address of the Dispute module that disputeId maps to
+   * @calls mapping(uint256 disputeId => Dispute dispute) public disputes;
+   */
+  public async readDisputes({ disputeId }: { disputeId: number | string | bigint }) {
```

Comment:
I think if we want to support read functions, we need to define and export the return types. Otherwise users have to figure out the types and define the types in their application

Reply:
I feel that based on the user story we created, these read functions are not important, except readDisputes, which is supported by API. 
It's better for us to keep the SDK lean and make it easy for users by just saying SDK is for writing and API is for reading.